index.html

```diff
@@ -388,9 +388,101 @@ <h2 id="protocol-registry">
     <h3>
       Inclusion criteria
     </h3>
+    <aside class="note">
+      The below criteria are a work in progress and are likely to change as
+      this document evolves.
+    </aside>
+    <p>
+      To be included in the registry, the [=digital credential/exchange
+      protocol=]:
+    </p>
+    <ol>
+      <li>MUST be standardized at a recognized standards organization and have
+      a stable URL.
+      </li>
+      <li>MUST define a [[WebIDL]] [=dictionary=] representation of the
```

Comment:
This seems like a good goal to me, but seems hard to accomplish logistically. Where does the WebIDL live? In this spec? In the protocol spec? What happens when they get out of sync?

Reply:
I was imagining the protocol spec. In our spec, we would only link to the dictionary (i.e., or spec would only have `[FooBar](https:/link-to-whatever.com/spec/#FooBar)`).

Reply:
It seems like it would be easier to include the definition in this spec as part of the inclusion process.

Reply:
It could certainly be done that way. I do share @samuelgoto’s concerns about this spec falling out of sync though. 

I was thinking that having just the name of dictionary in this spec minimizes the chance of things getting out of sync. Listing just the name is also what the Cred Man spec does in its registry:

https://www.w3.org/TR/credential-management-1/#credential-type-registry-appropriate-interface-object

Reply:
Either would work, but we shouldn't require other specs that may already be completed to reopen themselves if they could just drop some webidl that describes themselves here. Particularly where they don't already define their structures with webidl.

Reply:
Yes, that's a good point. Someone can correct me, but I think the only candidates right now are OpendID4VP (and the mDoc part), and maybe W3C VC? I think all of those are in active development with the aim of integrating with this.

> but we shouldn't require other specs that may already be completed to reopen themselves if they could just drop some webidl that describes themselves here. 

Generally speaking, what I'm finding is that you can't take existing specs and expect them to work out of the box with the DC API. They really need to be developed in tandem because the Web and JS have very specific requirements. The IDL requirement, along with clearly specified validation criteria, is actually a forcing function for those specs to integrate properly with this spec.

My feeling right now is that they might need to reopen themselves if they want to integrate, because they would't of being designed to work with a JS API in the first place (and if any such spec exists).

Reply:
I expanded on my concerns about this specific requirement here on this other PR, but I remain fairly concerned (security-wise) about requiring all browsers to redeploy before protocols can be extended:

https://github.com/WICG/digital-credentials/pull/156#discussion_r1727745625